appformer-kogito-bridge/src/main/java/org/appformer/kogito/bridge/client/resource/producer/ResourceClientServiceProducer.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.appformer.kogito.bridge.client.resource.producer;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+
+import org.appformer.kogito.bridge.client.resource.ResourceContentService;
+import org.appformer.kogito.bridge.client.resource.impl.MockResourceContentService;
+import org.appformer.kogito.bridge.client.resource.impl.ResourceContentServiceImpl;
+
+/**
+ * Produces {@link ResourceContentService} beans according to whether the envelope API is available or not
+ *
+ */
+public class ResourceClientServiceProducer {
+
+    @Produces
+    @ApplicationScoped
+    public ResourceContentService produce() {
+        boolean isResourceContentServiceAvailable = isResourceContentServiceAvailable();
+        if (isResourceContentServiceAvailable) {
+            return new ResourceContentServiceImpl();
+        } else {
+            return new MockResourceContentService();
+        }
+    }
+
+    private native boolean isResourceContentServiceAvailable()/*-{
```

Comment:
I think we should prevent this as much as possible - having GWT native interfaces and code here.... hard to expose and maintain long term...

Reply:
I completely agree, I changed to use jsinterop instead